packages/origin-organization-irec-api/src/connection/handlers/create-connection.handler.ts

```diff
@@ -0,0 +1,42 @@
+import { CommandHandler, EventBus, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { BadRequestException } from '@nestjs/common';
+import { IRECAPIClient } from '@energyweb/issuer-irec-api-wrapper';
+import { CreateConnectionCommand } from '../commands/create-connection.command';
+import { Connection } from '../connection.entity';
+import { ConnectionDTO } from '../dto/connection.dto';
+import { ConnectionCreatedEvent } from '../events/connection-created.event';
+import { RegistrationService } from '../../registration';
+
+@CommandHandler(CreateConnectionCommand)
+export class CreateConnectionHandler implements ICommandHandler<CreateConnectionCommand> {
+    constructor(
+        @InjectRepository(Connection)
+        private readonly repository: Repository<Connection>,
+        private readonly registrationService: RegistrationService,
+        private readonly eventBus: EventBus
+    ) {}
+
+    async execute({
+        user: { organizationId },
+        credentials: { userName, password, clientId, clientSecret }
+    }: CreateConnectionCommand): Promise<ConnectionDTO> {
+        const client = new IRECAPIClient(process.env.IREC_API_URL);
```

Comment:
@piotr @JosephBagaric Guys, do you have any ideas on how to mock IRECAPIClient for a test?

Reply:
1. Make it a dependency instead of creating an instance
2. Use override feature in the test similarly to https://github.com/energywebfoundation/origin/blob/ec5aa08791af0abfe3b1ee059657518dff68c05e/packages/exchange/test/exchange.ts#L146-L147